cumulusci/core/flowrunner.py

```diff
@@ -140,13 +160,32 @@ def __repr__(self):
         )
 
 
-StepResult = namedtuple(
-    "StepResult",
-    ["step_num", "task_name", "path", "result", "return_values", "exception"],
-)
+class StepResult:
+    step_num: StepVersion
+    task_name: str
+    path: str
+    result: Any
+    return_values: Any
+    exception: Optional[Exception]
+
+    def __init__(
+        self,
+        step_num: StepVersion,
+        task_name: str,
+        path: str,
+        result: Any,
+        return_values: Any,
+        exception: Optional[Exception],
+    ):
+        self.step_num = step_num
+        self.task_name = task_name
+        self.path = path
+        self.result = result
+        self.return_values = return_values
+        self.exception = exception
```

Comment:
↔️: [`typing.NamedTuple`](https://docs.python.org/3.8/library/typing.html#typing.NamedTuple)
 ```python
class StepResult(typing.NamedTuple):
    step_num: StepVersion
    task_name: str
    path: str
    result: Any
    return_values: Any
    exception: Optional[Exception]
```

Reply:
TIL. That's clean. Will use that approach to avoid adding code.